engines/registration_sharing/app/controllers/registration_sharing/rmt_to_rmt_controller.rb

```diff
@@ -34,7 +34,7 @@ def destroy
     protected
 
     def system_params
-      params.permit(:login, :password, :hostname, :proxy_byos_mode, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data)
+      params.permit(:login, :password, :hostname, :proxy_byos, :proxy_byos_mode, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data)
```

Comment:
What happens when we send `proxy_byos_mode` from a sibling that is in RMT 2.19 to a sibling that is on RMT 2.17? On 2.17 `proxy_byos_mode` does not exist in the DB.

Reply:
we can [adapt/update](https://github.com/SUSE/rmt/pull/1194#discussion_r1699604905) the code in the regsharing engine to handle that scenario

```ruby
    def system_params
      if System.column_names.include?('proxy_byos_mode') && params.include?('proxy_byos')
        # read the old value and convert it to the new value
        foo = params[:proxy_byos]
        params[:proxy_byos_mode] = <correct_value_from_foo>
        params.permit(:login, :password, :hostname, :proxy_byos_mode, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data) 
      elsif System.column_names.include?('proxy_byos') && params.include?('proxy_byos_mode')
        # update accordingly to proxy_byos
        params.permit(:login, :password, :hostname, :proxy_byos, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data)
      else # proxy_byos -> proxy_byos case
        ...
      else # proxy_byos_mode -> proxy_byos_mode
      end
    end
```
With that we can remove the column at this time, in any case, we would need another release to update this part
I do not have strong opinions on the steps to remove the column, maybe keeping the column and removing it in a different RMT version when all siblings have `proxy_byos_mode` is simpler

Reply:
I'd suggest to stay compatible with registration sharing with RMTs < 2.19, the registration sharing will need to send both parameters (`proxy_byos` + `proxy_byos_mode`) and be able to receive both parameters (`proxy_byos` + `proxy_byos_mode`) to cover both directions of sync between new + old RMTs: 
RMT 2.17 -> RMT 2.19, RMT 2.19 -> RMT 2.17

Once all pubcloud RMTs are upgraded, we can remove all usage of `proxy_byos` + also drop the column.

Reply:
yes, I agree with that, and I have kept the `proxy_byos` column (no more `remove` migration code), regsharing code needs to be update to include that column, and depending on the enum we need to update the actual migration

Reply:
Sorry, I still do not follow. 

> regsharing code needs to be update to include that column,

yes the regsharing code needs to be updated, but it cannot be updated retroactively into RMT < 2.19. If I understand it correctly, at present, if RMT 2.19 gets deployed and we send along the new entry then RMT < 2.19 will blow up because the DB entry does not yet exist.

What am I missing?

Reply:
In the scenario of the migration removing the `proxy_byos` column on RMT 2.19:
Yes, but when the params for regsharing are sent, the RMT 2.19 can send the two (`proxy_byos` and `proxy_byos_mode` even if the column does not exist in current server, we know the values it should have

in the receiving RMT < 2.19 we can check if the system instance has `proxy_byos` or `proxy_byos_mode` and the content of the params and act accordingly

So, it would work but, as mentioned in a different comment, we would need an extra step to remove that handling logic, hopefully that question/topic is resolved now


That said, this PR does not remove the `proxy_byos` column on this step as we are not saving any step doing that in this PR and seems simpler that handling the logic stated above

Reply:
How about adding tests for a 2.19 system sharing with <2.19, and vice versa? This should clarify the behavior.

Reply:
> in the receiving RMT < 2.19 we can check if the system instance has proxy_byos or proxy_byos_mode and the content of the params and act accordingly

How? We cannot do anything that is not already there, the code for RMT < 2.19 is frozen....

Anyone else getting an "oh shit" moment? Or am I really missing something that is obvious to everyone else?

Reply:
OK, so RMT < 2.19 is frozen, on RMT < 2.19 we can send _whatever_ we want, let's say we send `proxy_byos` **and** `proxy_byos_mode` (from an RMT 2.19, for example) then [we allow some params to be present](https://github.com/SUSE/rmt/blob/56ed01ffc0e88f8bc48dd453dd27819cd1209d0f/engines/registration_sharing/app/controllers/registration_sharing/rmt_to_rmt_controller.rb#L37) meaning....the info on RMT < 2.19 is OK

did I explain it better this time ?

Now if we send the the info from RMT < 2.19 to RMT 2.19 [we have a logic to handle the missing param](https://github.com/SUSE/rmt/blob/master/engines/registration_sharing/app/controllers/registration_sharing/rmt_to_rmt_controller.rb#L14) `proxy_byos_mode`, as in to include the value into the `system` being created as a consecuence of regsharing